Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -978,12 +978,12 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Node>> Node::clone_node(Document* document,
         // Set copy’s namespace, namespace prefix, local name, and value to those of node.
         auto& attr = static_cast<Attr&>(*this);
         copy = attr.clone(*document);
-    } else if (is<Text>(this)) {
+    } else if (is<Text>(this) || is<CDATASection>(this)) {
```

Comment:
A `CDATASection` doesn't pass the `is<Text>` check? I would expect that to pass since `CDATASection` inherits from `Text`

Reply:
Oh, the `fast_is` impl for text only checks `TEXT_NODE` specifically. Kinda surprising behavior, because I'm pretty sure the `dynamic_cast` impl would pass :thinking:

Reply:
The reason it doesn't pass the check is because of this line:

https://github.com/LadybirdBrowser/ladybird/blob/a4b289ebac837c4cfeced9b9145aa5e15cb3281e/Userland/Libraries/LibWeb/DOM/Text.h#L67

This cases `Node::is_text()` only returns true for `Text` nodes and nothing else. 

I think the best thing to do would be to change `Node::is_text()` to include all text node types, but then I'd have to go through and audit usages of `is<Text>(node)` and `node.is_text()` to ensure that nothing breaks. I know there are at least a couple of places that rely on the current behavior.

If we don't want to change `Node::is_text()` I think we should at least change `fast_is<Text>()` to ensure the behavior of `is<Text>()` is consistent with how `is<T>()` behaves everywhere else.

Reply:
Right, I don't think we need to go through that headache for this PR. Maybe just add a FIXME?

Reply:
I've added a comment above this line for now. I'll open another PR to sort this out once I've gone through all existing usages of `is<Text>(node)` and `node.is_text()`